### PR TITLE
[4/5] Properly enable GC for conda refactor

### PR DESCRIPTION
**Pull request by likawind**
## Describe your changes and why you are making these changes
This PR enables proper GC behavior after conda refactor where all function operators are not connected to an `exec_env` object. As a result, we modified the existing query to use `engine_config` instead to identify inactive conda env names.

Since we are also deprecating the 'garbage_collected' field, we do not rely on this field to check if an conda env is active. Instead, we simply use `conda env list` and check if a desired env is already created.

This PR also include some minor bug fixes in previous PRs to ensure all conda features are working properly.

## Related issue number (if any)
ENG-2467

## Loom demo (if any)
Tested performed:
unit test for new query
create and execute workflows without conda
create / update, and execute workflows with conda

Not tested:
data migration, which will include in follow-up PR.

## Checklist before requesting a review
- [ ] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [ ] I have performed a self-review of my code.
- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [ ] I have run the integration tests locally and they are passing.
- [ ] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [ ] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: manual_qa_tests/workflows/succeed_complex.py [hunk 1]
```diff
@@ -12,7 +12,7 @@
 * Data Page: pred_churn artifact should appears."""
 
 
-@aq.op
+@aq.op(requirements=["numpy"])
```
Comment (likawind) [inline manual_qa_tests/workflows/succeed_complex.py:15]: required for conda to avoid `pip freeze` behavior

File: sdk/aqueduct/constants/enums.py [hunk 2]
```diff
@@ -193,6 +193,7 @@ class ExecutionMode(str, Enum, metaclass=MetaEnum):
 
 class RuntimeType(str, Enum, metaclass=MetaEnum):
     AQUEDUCT = "aqueduct"
+    AQUEDUCT_CONDA = "aqueduct_conda"
```
Comment (likawind) [inline sdk/aqueduct/constants/enums.py:196]: required for SDK to fetch operators with `aq_conda` configs

File: src/golang/cmd/server/handler/preview.go [hunk 3]
```diff
@@ -356,12 +356,14 @@ func setupCondaEnv(
 		}
 
 		op, ok := dagSummary.Dag.Operators[opId]
-		if ok && op.Spec.IsFunction() {
-			engineConfig := op.Spec.EngineConfig()
-			engineConfig.Type = shared.AqueductCondaEngineType
-			engineConfig.AqueductCondaConfig = &shared.AqueductCondaConfig{
-				Env: env.Name(),
-			}
+		if ok && op.Spec.HasFunction() {
```
Comment (likawind) [inline src/golang/cmd/server/handler/preview.go:359]: use `hasFunction` to cover metrics and checks

File: src/golang/cmd/server/handler/preview.go [hunk 4]
```diff
@@ -356,12 +356,14 @@ func setupCondaEnv(
...
+			op.Spec.SetEngineConfig(&shared.EngineConfig{
+				Type: shared.AqueductCondaEngineType,
+				AqueductCondaConfig: &shared.AqueductCondaConfig{
+					Env: env.Name(),
+				},
+			})
+			dagSummary.Dag.Operators[opId] = op
```
Comment (likawind) [inline src/golang/cmd/server/handler/preview.go:366]: The prev implementation was wrong and did not update the object inside the map

**Review by cw75 — APPROVED** [review]
LGTM

**Pull request merged**